https.c

```diff
@@ -114,7 +114,11 @@ _SSL_check_server_cert(SSL *ssl, const char *hostname)
                 
                 for (i = 0; i < n && match != 1; i++) {
                         altname = sk_GENERAL_NAME_value(altnames, i);
+#if OPENSSL_VERSION_NUMBER >= 0x1010000fL
+                        p = (const char *) ASN1_STRING_get0_data(altname->d.ia5);
```

Comment:
I'm wondering if we'll want to wrap this in a function.
I don't know how often we'll need to use the `ASN1_STRING_data` but not having to use an if else every time would be good

Reply:
Seems like a wrapper function could work for this as well. I'll add one and ditch the macros.